ts/src/core/ent.ts

```diff
@@ -949,8 +954,19 @@ export function buildGroupQuery<T extends Data = Data, K = keyof T>(
 
   // window functions work in sqlite!
   //    https://www.sqlite.org/windowfunctions.html
+  const fieldString = fields
+    .map((f) => {
+      if (typeof f === "object") {
+        // TS doesn't understand that K can only be a string, so we need
+        // for it to treat f as the object we know it is.
+        const fObj = f as { alias: string; column: string };
+        return `${fObj.column}.${fObj.alias}`;
```

Comment:
this one seems broken

Reply:
Yeah I had trouble with Typescript here. `Data` is defined as `{ [key: string]: any }`, so when `K` is defined as `keyof T` (and `T` extends `Data`), then `K` must be a `string`. But for some reason, `fields` gets typed as `K | string | { alias: string; column: string }`, and when I do `typeof f === "object"` it only changes the type to `K | { alias: string; column: string }`. Even though `K` is a string. 😕🤷🏼‍♂️

Reply:
Mostly referring to alias and column being flipped

Reply:
![owl-puppet](https://github.com/lolopinto/ent/assets/974639/2f7e7daf-b34b-40d1-a838-c088297d52a7)